packages/contracts/contracts/ProxyAdmin.sol

```diff
@@ -1,28 +1,169 @@
+     *
+     * @param _proxy     Address of the proxy, which can be a default proxy or a non-standard proxy.
+     * @param _proxyType The proxy's type. This is the zero-address for default proxies.
+     * @param _code      Creation bytecode to be deployed.
+     */
     function setProxyCode(
-        address _proxy,
+        address payable _proxy,
         bytes32 _proxyType,
-        bytes memory _data
-    ) public {}
+        bytes memory _code
+    ) public {
+        // If no proxy type has been set, use the default proxy.
+        if (_proxyType == bytes32(0)) {
+            // Upgrade the proxy's implementation to be the ProxyUpdater, and call `setCode` on the
+            // proxy.
+            Proxy(_proxy).upgradeToAndCall(
+                proxyUpdater,
+                abi.encodeWithSignature("setCode(bytes32,bytes)", EIP1967_IMPLEMENTATION_KEY, _code)
+            );
+        } else {
+            // If a proxy type has been specified for this action, then the ProxyAdmin delegatecalls
+            // the corresponding adapter, which contains the logic that will be used to call the
+            // proxy from the context of this contract.
+
+            // Get the adapter that corresponds to this proxy type.
+            address adapter = registry.adapters(_proxyType);
+            require(adapter != address(0), "ProxyAdmin: proxy type has no adapter");
+
+            // Delegatecall the adapter to upgrade the proxy's implementation to be the
+            // ProxyUpdater, which has the `setCode` function.
+            _upgradeWithAdapter(_proxy, adapter, proxyUpdater);
+
+            // Delegatecall the adapter, which in turn will call the proxy to trigger a `setCode`
+            // action.
+            (bool success, ) = adapter.delegatecall(
+                abi.encodeWithSignature("setProxyCode(address,bytes)", _proxy, _code)
+            );
+            require(success, "ProxyAdmin: delegatecall to set proxy code failed");
+        }
+    }
 
+    /**
+     * @notice Modifies a storage slot within the proxy contract.
+     *
+     * @param _proxy     Address of the proxy, which can be a default proxy or a non-standard proxy.
+     * @param _proxyType The proxy's type. This is the zero-address for default proxies.
+     * @param _key       Storage key to modify.
+     * @param _value     New value for the storage key.
+     */
     function setProxyStorage(
-        address _proxy,
+        address payable _proxy,
         bytes32 _proxyType,
         bytes32 _key,
         bytes32 _value
-    ) public {}
+    ) public {
+        // If no proxy type has been set, use the default proxy.
+        if (_proxyType == bytes32(0)) {
+            // Upgrade the proxy's implementation to be the ProxyUpdater, and call `setStorage` on
+            // the proxy.
+            Proxy(_proxy).upgradeToAndCall(
+                proxyUpdater,
+                abi.encodeWithSignature("setStorage(bytes32,bytes32)", _key, _value)
```

Comment:
I think it would be worth adding an interface for `IProxyAdapter`. Will simplify the code quite a bit by making it possible to do `abi.encodeCall(IProxyAdapter.setStorage, _key, _value)` which will also give you type safety.